src/lib/trakeva_cache.ml

```diff
@@ -0,0 +1,195 @@
+(**************************************************************************)
+(*  Copyright 2014, Sebastien Mondet <seb@mondet.org>                     *)
+(*                                                                        *)
+(*  Licensed under the Apache License, Version 2.0 (the "License");       *)
+(*  you may not use this file except in compliance with the License.      *)
+(*  You may obtain a copy of the License at                               *)
+(*                                                                        *)
+(*      http://www.apache.org/licenses/LICENSE-2.0                        *)
+(*                                                                        *)
+(*  Unless required by applicable law or agreed to in writing, software   *)
+(*  distributed under the License is distributed on an "AS IS" BASIS,     *)
+(*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or       *)
+(*  implied.  See the License for the specific language governing         *)
+(*  permissions and limitations under the License.                        *)
+(**************************************************************************)
+
+open Nonstd
+open Pvem_lwt_unix
+open Pvem_lwt_unix.Deferred_result
+module String = StringLabels
+
+let debug = ref false
+
+let dbg fmt = ksprintf (eprintf "Trakeva_cache: %s\n%!") fmt
+    
+module Greedy_cache = struct
```

Comment:
Yes it grows without bounds and it eats a lot at once.
I use [greedy](http://dictionary.reference.com/browse/greedy?s=t) a synonym of “gluttonous.”